packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -735,6 +737,14 @@ export const getTeamsUnreadStatuses: (state: GlobalState) => [Set<Team['id']>, M
                     teamMentionsMap.set(channel.team_id, unreadCountObjectForChannel.mentions + previousMentionsInTeam);
                 }
             }
+
+            // Add has urgent mentions count from channel membership
+            if (unreadCountObjectForChannel.hasUrgent) {
+                const previousHasUrgetInTeam = teamHasUrgentMap.has(channel.team_id) ? teamHasUrgentMap.get(channel.team_id) as boolean : false;
+                if (!previousHasUrgetInTeam) {
+                    teamHasUrgentMap.set(channel.team_id, unreadCountObjectForChannel.hasUrgent);
+                }
```

Comment:
I feel like the actual condition should be sufficient for this

```suggestion
                if (!teamHasUrgentMap.has(channel.team_id)) {
                    teamHasUrgentMap.set(channel.team_id, unreadCountObjectForChannel.hasUrgent);
                }
```

Reply:
yeah, but this `teamHasUrgentMap.has(channel.team_id)` should give you the same value as this `teamHasUrgentMap.get(channel.team_id) as boolean`

When I get this right you check for the existence of the tea_id in the map and if not present you set the value for it, right? If that's the case you can just do this and your good to go, since `get` called on a key that does not exist just returns `undefined`

```
if (!teamHasUrgentMap.get(channel.team_id)) {
    teamHasUrgentMap.set(channel.team_id, unreadCountObjectForChannel.hasUrgent);
}
```

Or is there something I am missing?

Reply:
Maybe I'm missing something, how can `teamHasUrgentMap.has(team_id)` be the same with `teamHasUrgentMap.get(team_id)`? 

Let's say the first pass it doesn't exist so it sets it and the value is `false` for `isUrgent` the second pass it exists so `has` returns true so it won't set it again.

I can see just doing the following though:

```ts
if (unreadCountObjectForChannel.hasUrgent) {
    teamHasUrgentMap.set(channel.team_id, true);
}
```

Reply:
> @michelengelen maybe I'm missing something, how can `teamHasUrgentMap.has(team_id)` be the same with `teamHasUrgentMap.get(team_id)`?
> 
> Let's say in the first pass it doesn't exist so it sets it and the value is `false` for `isUrgent`, in the second pass it exists so `has` returns true so it won't set it again.
> 
> I can see just doing the following though:
> 
> ```ts
> if (unreadCountObjectForChannel.hasUrgent) {
>     teamHasUrgentMap.set(channel.team_id, true);
> }
> ```

Sorry for answering this so late, I have been on PTO. 🙇🏼 

I think we are talking about different things here. I am not concerned by the initial check for `unreadCountObjectForChannel`, but rather the ternary that follows that 

```ts
const previousHasUrgetInTeam = teamHasUrgentMap.has(channel.team_id) ? teamHasUrgentMap.get(channel.team_id) as boolean : false;
```

This condition can be simplified, since you init `teamHasUrgentMap` yourself as an empty `Map`, so there is no outside influence. The type for that also indicates, that there can only be `boolean` values in the properties. 

That means everything you check against is sane by default.

Considering that running `get` on the `Map` is essentially the same as using `has`, since the only values it returns are ...
- `undefined` (when `team_id` is not part of the `Map`)
- `false`
- `true`

So in an effort to reduce complexity you can just remove the call to `has` on `teamHasUrgentMap`.

I am not sure what you meant by the default value, since `get` does not have that behavior.